python/pyspark/sql/tests/test_functions.py

```diff
@@ -286,6 +289,42 @@ def test_dayofweek(self):
         row = df.select(dayofweek(df.date)).first()
         self.assertEqual(row[0], 2)
 
+    # Test added for SPARK-37738; change Python API to accept both col & int as input
+    def test_date_add_function(self):
+        dt = datetime.date(2021, 12, 27)
+        df = self.spark.createDataFrame([Row(date=dt, add=2)])
```

Comment:
How about we add schema here

```python
df = self.spark.createDataFrame([Row(date=dt, add=2)], "date date, add integer")
```

Reply:
I didn't know you could specify schema input like this- have amended; thank you!